### ISSUE TITLE
Question about build_dataset function

### ISSUE THREAD

**Issue by robbine** [open-mmlab/mmclassification#718]
`    datasets = [build_dataset(cfg.data.train)]
    if len(cfg.workflow) == 2:
        val_dataset = copy.deepcopy(cfg.data.val)
        val_dataset.pipeline = cfg.data.train.pipeline
        datasets.append(build_dataset(val_dataset))`

 The code snippet above was found in tools/train.py.
I cound not figure out why  **val_dataset.pipeline** is set to **cfg.data.train.pipeline**
Is there some special reason about this design except for concerning about efficiency issues ?

Thanks a lot !

**Comment by Ezra-Yu** [comment]
You asked a very detailed question.

Here is the [guide](https://mmclassification.readthedocs.io/en/latest/tutorials/runtime.html#customize-workflow) about the workflow.

notice that :
***Therefore, the only difference between [('train', 1), ('val', 1)] and [('train', 1)] is that the runner will calculate losses on the validation set after each training epoch.***
You can see `workflow:val` as a tool to check the losses in val, so here we use `train.pipeline`.  Actually the design of workflow is a little confusing.

The evaluation in the validation dataset is in the [`EvalHook`](https://mmclassification.readthedocs.io/en/latest/tutorials/runtime.html#hooks), which will use the `test.pipeline`.

**Comment by robbine** [comment]
I'll look into it.
Thanks for your detailed explanation.
